### ISSUE TITLE
Some dual-carriageway intersections are collapsed into a single point

### ISSUE THREAD

**Issue by seav** [maning/osmphgps#48]
Look at the intersection of Paseo de Roxas in Makati with the following streets in the Garmin map:
- Dela Rosa  (OSM: http://osm.org/go/4zhFr0aG9--?m)
- Legazpi  (OSM: http://osm.org/go/4zhFrPzcz--?m)

You will notice the dual carriageway nature of Paseo de Roxas on these intersections collapse into a single point. This results into wrong routing because the linking way on Dela Rosa is part of a turn restriction while the linking way on Legazpi is oneway. This results into the following illegal routes:
- The map can now tell you to take a U-turn from Paseo de Roxas northbound to southbound at the Dela Rosa intersection
- The map can now tell you to take a left turn from Paseo de Roxas northbound to Dela Rosa
- The map can now tell you to take a left turn from Legazpi coming from the east to Paseo de Roxas southbound

I thought that this was just an issue with Garmin's reduced spatial resolution (it seems to store less precision in coordinates to save space), but if you'll notice, the traffic lights POIs are still in the correct position. So either Garmin stores _even less_ precision in coordinates for roads, or there is something wrong in the conversion process.

I noticed this using an older version of the Garmin map, but the problem still exists using the latest Garmin map (May 9, 2012).

**Comment by maning** [comment]
I removed the `remove-short-arcs` directive, please test the following files:
- mapsource installer - http://db.tt/YKeCp6pA
- gmapsupp.img - http://db.tt/bkrW6pRa

**Comment by seav** [comment]
I think that fixed it. The ways are no longer collapse.

BTW, do turn restrictions with a way via work? (Separate issue.)

**Comment by maning** [comment]
Does it solve the routing issue? AFAIK, turn restrictions are supported but not for every possible variation of the schema.

**Comment by seav** [comment]
Yes, it fixed the routing issue due to the missing way. The following example is now fixed:
- The map tells you to take a left turn from Legazpi coming from the east to Paseo de Roxas southbound which it should not

The following 2 issues are not fixed:
- The map tells you to take a U-turn from Paseo de Roxas northbound to southbound at the Dela Rosa intersection which it should not
- The map tells you to take a left turn from Paseo de Roxas northbound to Dela Rosa which it should not

It's because it seems turn restrictions where the via is a way does not seem to be honored. I've looked at an example along Macapagal Blvd and the map gives wrong U-turn instructions on a no-U-turn intersection.
